src/util/type.cpp

```diff
@@ -28,3 +28,9 @@ void type_with_subtypest::move_to_subtypes(typet &type)
   sub.push_back(static_cast<const typet &>(get_nil_irep()));
   sub.back().swap(type);
 }
+
+typet remove_const(typet type)
```

Comment:
const argument? Or if this is modifying the argument, then why return ?

Reply:
It's returning a copy with the modification. If the argument was const we'd have to make two copies, if it was const reference we'd still have a copy so we don't gain anything in either case.

Reply:
OK, I like the const just to make it explicit - and of course depending on how much this is called, an extra copy probably isn't the end of the world... But its a nitpick really on my part.